---
title: "Analysis of Genetic Data 1: Inferring Population Structure"
author: Peter Carbonetto
output:
 beamer_presentation:
   template: beamer.tex
   keep_tex: false
---

Download 1000 Genomes data
==========================

We download the 1000 Genomes data from the European Bioinformatics
Institute. Downloading may take 10--20 minutes.

+ Long URL: ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/supporting/hd_genotype_chip

+ Short URL: http://bit.ly/2G7ZWYu
  
+ Download this file into the `data` folder: 
  `ALL.chip.omni_broad_sanger_combined.
   20140818.snps.genotypes.vcf.gz`

On the RCC cluster, you can run these commands:

```{bash, eval=FALSE}
cd data
wget http://bit.ly/2C617oO -O 1kg.vcf.gz
```

Explore the VCF file
====================

Let's run some simple shell commands to inspect the genotype data
stored in the VCF file.

```{bash eval=FALSE}
cd data
ls -lh 1kg.vcf.gz
zcat 1kg.vcf.gz | less -S // On Mac, use gzcat.
```

+ VCF reference: http://www.cog-genomics.org/plink2/formats#vcf

VCF files: concepts
===================

+ The Variant Call Format (VCF) is a text format for storing many
types of DNA variant data (e.g., SNPs, deletions, insertions), and
for annotating these variants.

+ It is one of the most commonly used data formats in genetics.

+ It is an inefficient way to store genotype data (this is why we
have compressed it).

+ See also:
    - https://vcftools.github.io 
    - https://samtools.github.io/hts-specs 
    - doi:10.1093/bioinformatics/btr330

Download PLINK
==============

We download the stable version of the PLINK software in the `bin
folder`.

+ URL: http://www.cog-genomics.org/plink2

On the RCC cluster, you can run these commands to download and test
the 64-bit Linux binary for PLINK 1.9b5.2:

```{bash, eval=FALSE}
cd bin
wget http://bit.ly/2nUxOQX -O plink.zip
unzip plink.zip
./plink --version
```

Convert VCF to PLINK
====================

Run this command to convert the genotypes from VCF to the PLINK text
format. This may take a few minutes to complete.

```{bash, eval=FALSE}
cd data
../bin/plink --vcf 1kg.vcf.gz --recode \
  --chr 1-22 --allow-extra-chr \
  --geno 0.01 --out 1kg
```

+ Command details:
    - Creates two new files: `1kg.map` and `1kg.ped`.
    - We retain only SNPs on (autosomal) chromosomes 1--22.
    - We remove any SNPs with >1% missing genotypes.
    - These steps are taken to simplify the analyses.

Explore PLINK files
===================

Let's run some simple shell commands to inspect the 1000 Genomes
genotype data stored in the PLINK files.

```{bash, eval=FALSE}
head 1kg.map
tail 1kg.map
wc -l 1kg.map
less -S 1kg.ped
wc -l -w 1kg.ped
```

Next, use these same commands to inspect the AffyMetrix Human Origins
data stored in `origins.map` and `origins.ped` (they are already
included in the git repository).

+ Columns in `.map` file: (1) chromosome; (2) marker id; (3)
genetic distance on chromosome, in cM; (4) base-pair position on
chromosome.
  
+ Columns in `.ped` file: (1) family id, (2) individual id, (3) father
id, (4) mother id, (5) gender, (6) phenotype measurement, (7---) SNP
genotypes.

PLINK files: concepts
=====================

+ The most commonly used format for storing human genoytpe data.

+ Less flexible than VCF.

+ The PLINK text format is easy to view and manipulate with simple
shell commands (e.g., `wc`, `grep`, `cat`, `cut`, `paste`).

+ For long-term storage, use PLINK binary (`.bed`) format. It is much
more efficient, but is not human readable.

+ See: http://www.cog-genomics.org/plink/1.9/formats#ped

Prepare the 1000 Genomes data
=============================

To speed up the data processing steps, we convert to binary PLINK
format, then we remove related samples (which standard population
structure analyses are not designed to handle).

1. Convert to PLINK binary format:
```
../bin/plink --file 1kg --make-bed --out 1kg
```

2. Inspect the 3 new files: `1kg.bed`, `1kg.bim`, `1kg.fam`.

3. Remove 29 of 31 related samples:
```
cut -f 1 20140625_related_individuals.txt \
  > temp.txt
paste temp.txt temp.txt > samples.txt
../bin/plink --bfile 1kg --make-bed \
   --remove samples.txt --out 1kg_unrelated
```

Merge with Affymetrix Human Origins data
========================================

* Commands to merge the two data sets:
    ../bin/plink --bfile 1kg_unrelated --extract 1kg_origins_markers.txt \
      --make-bed --out 1kg_unrelated_reduced
    ../bin/plink --bfile origins --extract 1kg_origins_markers.txt \
      --remove dupids.txt --make-bed --out origins_reduced
    ../bin/plink --bfile 1kg_unrelated_reduced --bmerge origins_reduced \
      --out 1kg_origins_combined
* "Prune" SNPs in LD (typically you will want to be more aggressive):
    ../bin/plink --bfile 1kg_origins_combined --indep-pairwise 1000 500 0.8
    ../bin/plink --bfile 1kg_origins_combined --make-bed \
      --extract plink.prune.in --out 1kg_origins_pruned


# Rick Astley's Re-election Platform

Rick Astley's campaign promises:

- Never gonna give you up.
- Never gonna let you down.
- Never gonna run around and desert you.
- Never gonna make you cry.
- Never gonna say goodbye.
- Never gonna tell a lie and hurt you.

Whereas these pledges conform to the preferences of the **median voter**, we expect Congressman Astley to secure re-election.

# Caribbean Queen and Operation Urgent Fury

Billy Ocean released "Caribbean Queen" in 1984.

- Emphasized sharing the same dream
- Hearts beating as one

"Caribbean Queen" is about the poor execution of Operation Urgent Fury.

- Echoed JCS chairman David Jones' frustrations with military establishment.

Billy Ocean is advocating for what became the Goldwater-Nichols Act.

- Wanted to take advantage of **economies of scale**, resolve **coordination problems** in U.S. military.

# The Good Day Hypothesis

We know the following about Ice Cube's day.

1. The Lakers beat the Supersonics.
2. No helicopter looked for a murder.
3. Consumed Fatburger at 2 a.m.
4. Goodyear blimp: "Ice Cube's a pimp."

This leads to two different hypotheses:

- $H_0$: Ice Cube's day is statistically indistinguishable from a typical day.
- $H_1$: Ice Cube is having a good (i.e. greater than average) day.

These hypotheses are tested using archival data of Ice Cube's life.

# Example R code
# Example R stuff

```{r cars, echo = TRUE}
summary(cars)
```

# Slide with Plot

```{r pressure}
plot(pressure)
```

# ggplot code

```{r createdata, echo=TRUE, eval=FALSE}
df <- data.frame(x = rnorm(1000))
x <- df$x
base <- ggplot(df, aes(x)) + geom_density()  + scale_x_continuous(limits = c(-5, 5))
base + stat_function(fun = dnorm, colour = "red")
```


# Another Plot

```{r plotit, echo=FALSE}
library(ggplot2)
df <- data.frame(x = rnorm(1000))
x <- df$x
base <- ggplot(df, aes(x)) + geom_density()  + scale_x_continuous(limits = c(-5, 5))
base + stat_function(fun = dnorm, colour = "red")
```